### ISSUE TITLE
`go` version seems not to change with param `go-version`

### ISSUE THREAD

**Issue by marcello33** [Templum/govulncheck-action#16]
By observing the action's logs, apparently there's a difference in `golang` version between `GOLANG_VERSION=XXX` and `GoEnvironment Details: [...] Go-Version=goYYY`

This is an issue as it's impossible at the moment to test the code with the same version used in the repo itself, as the action param `go-version` seems to be ignored.

Latest go `v1.19.3` is always used, and this cause the tool not to detect some actual vulnerabilities.

**Comment by Templum** [comment]
Thanks for opening the issue, as soon as I find the time I will look into it. Perform some checks

**Comment by Templum** [comment]
@marcello33 I was able to give it a look and created branch `fix/16`. It turns out that this is just a display error, but let me explain.

I'm not a full-time golang developer hence my assumption was that information from the runtime package are taken at runtime (not at compile time). But turns out that is not the case.

Within my Dockerfile I use the builder pattern, ensuring the action itself is compiled with a golang version I control. The version chosen by the user of the action will be installed in the runtime part of the container, into which I also install the vulncheck cli. Meaning the CLI is run in the correct env, just the output is incorrect. In `fix/16` I switched to reading the go env. This contains the actual runtime information that is relevant to govulncheck. 

Before:
![image](https://user-images.githubusercontent.com/7703616/202467016-630bd9a0-b091-4ce3-8601-90cf83a0b2a3.png)

After:
![image](https://user-images.githubusercontent.com/7703616/202467136-07e0a2be-46fa-44a1-bee6-0cb8c190e503.png)
